src/models/partial_tx.ts

```diff
@@ -295,40 +369,69 @@ export class PartialTx {
       }
     });
     const tx = this.getTx();
-    const arr = ['PartialTx', tx.toHex(), ...changeOutputs];
+    // tokenData(1 byte), token(32 bytes | 1 byte), value(variable bytes with max 16)
+    const inputArr = this.inputs.map(i => [
+        Buffer.from([i.tokenData]).toString('hex'), // This ensures we always have 1 byte for token data
+        i.token,
+        i.value.toString(16),
+      ].join(''));
+    const arr = [
+      PartialTxPrefix,
+      tx.toHex(),
+      inputArr.join(':'),
+      changeOutputs.map(o => o.toString(16)).join(':'), // array of change outputs
+    ];
     return arr.join('|');
   }
 
   /**
    * Deserialize and create an instance of PartialTx
-   * This method is async because we need to fetch information on the inputs (token, value)
    *
    * @param {string} serialized The serialized PartialTx
    * @param {Network} network Network used when parsing the output scripts
    *
+   * @returns {PartialTx}
+   *
    * @throws {SyntaxError} serialized argument should be valid.
    * @throws {UnsupportedScriptError} All outputs should be P2SH or P2PKH
-   *
-   * @returns {Promise<PartialTx>}
    * @memberof PartialTx
    * @static
    */
-  static async deserialize(serialized: string, network: Network): Promise<PartialTx> {
+  static deserialize(serialized: string, network: Network): PartialTx {
 
     const dataArr = serialized.split('|');
-    const changeOutputs = dataArr.slice(2).map(x => +x);
+    const txHex = dataArr[1];
 
-    if (dataArr.length < 2 || dataArr[0] !== 'PartialTx' || !changeOutputs.every(x => Number.isInteger(x))) {
+    if (dataArr.length !== 4 || dataArr[0] !== PartialTxPrefix) {
       throw new SyntaxError('Invalid PartialTx');
     }
 
-    const txHex = dataArr[1];
+    const inputArr = dataArr[2].split(':').map(h => {
+      let it: number = 0;
```

Comment:
I had to pause for a while to understand this variable's meaning.
I suggest a small comment like this to help future readers:
```suggestion
      /**
       * Iterator index, keeps track of the current byte being read.
       */
      let it: number = 0;
```

Reply:
Added an explanation for this variable